dev-client/src/model/soilId/soilIdRanking.test.tsx

```diff
@@ -0,0 +1,298 @@
+    soilData: {
+      depthDependentData: [],
+    },
+    soilSeries: {
+      description: 'Description 1',
+      fullDescriptionUrl: 'Url 1',
+      name: 'Name 1',
+      taxonomySubgroup: 'Taxonomy subgroup 1',
+    },
+  },
+};
+const locationBasedSoil2 = {
+  dataSource: 'Data source 2',
+  distanceToNearestMapUnitM: 5,
+  match: {
+    rank: 2,
+    score: 0.5,
+  },
+  soilInfo: {
+    landCapabilityClass: {
+      capabilityClass: 'Class 2',
+      subClass: 'Subclass 2',
+    },
+    soilData: {
+      depthDependentData: [],
+    },
+    soilSeries: {
+      description: 'Description 2',
+      fullDescriptionUrl: 'Url 2',
+      name: 'Name 2',
+      taxonomySubgroup: 'Taxonomy subgroup 2',
+    },
+  },
+};
+
+const dataBasedSoil0 = {
+  dataSource: 'Data source 0',
+  distanceToNearestMapUnitM: 5,
+  combinedMatch: {
+    rank: 0,
+    score: 0.5,
+  },
+  dataMatch: {
+    rank: 1,
+    score: 0.5,
+  },
+  locationMatch: {
+    rank: 1,
+    score: 0.5,
+  },
+  soilInfo: {
+    landCapabilityClass: {
+      capabilityClass: 'Class 0',
+      subClass: 'Subclass 0',
+    },
+    soilData: {
+      depthDependentData: [],
+    },
+    soilSeries: {
+      description: 'Description 0',
+      fullDescriptionUrl: 'Url 0',
+      name: 'Name 0',
+      taxonomySubgroup: 'Taxonomy subgroup 0',
+    },
+  },
+};
+const dataBasedSoil1 = {
+  dataSource: 'Data source 1',
+  distanceToNearestMapUnitM: 5,
+  combinedMatch: {
+    rank: 1,
+    score: 0.5,
+  },
+  dataMatch: {
+    rank: 2,
+    score: 0.5,
+  },
+  locationMatch: {
+    rank: 2,
+    score: 0.5,
+  },
+  soilInfo: {
+    landCapabilityClass: {
+      capabilityClass: 'Class 1',
+      subClass: 'Subclass 1',
+    },
+    soilData: {
+      depthDependentData: [],
+    },
+    soilSeries: {
+      description: 'Description 1',
+      fullDescriptionUrl: 'Url 1',
+      name: 'Name 1',
+      taxonomySubgroup: 'Taxonomy subgroup 1',
+    },
+  },
+};
+const dataBasedSoil2 = {
+  dataSource: 'Data source 2',
+  distanceToNearestMapUnitM: 5,
+  combinedMatch: {
+    rank: 2,
+    score: 0.5,
+  },
+  dataMatch: {
+    rank: 0,
+    score: 0.5,
+  },
+  locationMatch: {
+    rank: 0,
+    score: 0.5,
+  },
+  soilInfo: {
+    landCapabilityClass: {
+      capabilityClass: 'Class 2',
+      subClass: 'Subclass 2',
+    },
+    soilData: {
+      depthDependentData: [],
+    },
+    soilSeries: {
+      description: 'Description 2',
+      fullDescriptionUrl: 'Url 2',
+      name: 'Name 2',
+      taxonomySubgroup: 'Taxonomy subgroup 2',
+    },
+  },
+};
+
+describe('location based matches', () => {
+  test('get top match by rank', () => {
+    const inputSoilIdResults = {
+      locationBasedMatches: [
+        locationBasedSoil0,
+        locationBasedSoil1,
+        locationBasedSoil2,
+      ],
+      dataBasedMatches: [],
+    };
+    expect(getTopMatch(inputSoilIdResults)).toEqual(locationBasedSoil0);
+  });
+
+  test('get top match by rank when reordered', () => {
```

Comment:
I think this test is redundant with the previous one; it's fine IMO to just have a single test verifying that out-of-order elements still get sorted/picked correctly.